src/unix/linux-core.c

```diff
@@ -241,6 +241,8 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
      * events, skip the syscall and squelch the events after epoll_wait().
      */
     if (uv__epoll_ctl(loop->backend_fd, op, w->fd, &e)) {
+      if (ENOMEM == errno)
+        return UV_ENOMEM;
```

Comment:
Wrong in that `w` has been removed from `loop->watcher_queue` but hasn't been activated.